### PR TITLE
[TD-12723]<feature>(query): add taos_result_block api for taosAdaptor ulimited pulling function

### PR DESCRIPTION
**Pull request by glzhao89**

### CODE REVIEW

**Review by huskar-t — COMMENTED** [review]
Need to add the api to the taos.def file

**Comment by glzhao89** [comment]
> 

added api in taos.def file

**Review by huskar-t — APPROVED** [review]
LGTM

File: src/client/src/tscSql.c [hunk 1]
```diff
@@ -547,6 +547,28 @@ int taos_fetch_block(TAOS_RES *res, TAOS_ROW *rows) {
...
+  SSqlRes *pRes = &pSql->res;
+
+  if (pCmd == NULL ||
+      pRes == NULL ||
+      pRes->qId == 0 ||
+      pRes->code == TSDB_CODE_TSC_QUERY_CANCELLED ||
+      pCmd->command == TSDB_SQL_RETRIEVE_EMPTY_RESULT ||
+      pCmd->command == TSDB_SQL_INSERT) {
+    return NULL;
+  }
```
Comment (shenglian-zhou) [inline src/client/src/tscSql.c:567]: when the fetch is initiated so that the result is filled?

**Pull request merged**